devcontainers/lib/dependabot/devcontainers/utils.rb

```diff
@@ -0,0 +1,24 @@
+# typed: false
+# frozen_string_literal: true
+
+module Dependabot
+  module Devcontainers
+    module Utils
+      def self.expected_config_basename(directory)
+        root_directory?(directory) ? ".devcontainer.json" : "devcontainer.json"
```

Comment:
I believe directory is `/` by default is users don't configure it. That means, Dependabot won't just work for users using `.devcontainer/devcontainer.json` and not explicitly configuring a directory. Correct?

Should we instead, given a directory, look for both `.devcontainer.json`, and `.devcontainer/devcontainer.json`, relative to the configured directory?

Reply:
That's correct, this implementation requires setting `directory`.

I should not there's a third, more general option for where a config lives, `.devcontainer/<SOME_FOLDER>/devcontainer.json`, where `SOME_FOLDER` can be anything.  

I am on the fence on what makes the most sense.  I asked in Slack and sounded like this was a natural pattern (to use `directory` to point to the appropriate config).  If we don't require `directory` to be set when discovering configs, we'll need to check all folders nested in `.devcontainer` too (probably not a huge deal).  It also complicated some of the other components, but I could probably work around that

Reply:
We chatted offline - sounds like this initial approach is OK. We can iterate if it becomes unclear.  

[ref](https://github.slack.com/archives/C03UT3Q6V34/p1701369938981659?thread_ts=1700257487.226549&cid=C03UT3Q6V34)